heron/metricscachemgr/tests/java/com/twitter/heron/metricscachemgr/metricscache/MetricsCacheQueryUtilsTest.java

```diff
@@ -0,0 +1,288 @@
+// Copyright 2016 Twitter. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.twitter.heron.metricscachemgr.metricscache;
+
+import java.util.AbstractMap;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import com.twitter.heron.metricscachemgr.metricscache.query.ExceptionDatum;
+import com.twitter.heron.metricscachemgr.metricscache.query.ExceptionRequest;
+import com.twitter.heron.metricscachemgr.metricscache.query.ExceptionResponse;
+import com.twitter.heron.metricscachemgr.metricscache.query.MetricDatum;
+import com.twitter.heron.metricscachemgr.metricscache.query.MetricGranularity;
+import com.twitter.heron.metricscachemgr.metricscache.query.MetricRequest;
+import com.twitter.heron.metricscachemgr.metricscache.query.MetricResponse;
+import com.twitter.heron.metricscachemgr.metricscache.query.MetricTimeRangeValue;
+import com.twitter.heron.proto.system.Common;
+import com.twitter.heron.proto.tmaster.TopologyMaster;
+
+import static com.twitter.heron.metricscachemgr.metricscache.MetricsCacheQueryUtils.toProtobuf;
+
+public class MetricsCacheQueryUtilsTest {
+  private static void assertMetricRequest(
+      MetricRequest request,
+      List<AbstractMap.SimpleEntry<String, List<String>>> componentNameInstanceId,
+      List<String> metricNames, MetricGranularity aggregationGranularity) {
+    Assert.assertEquals(aggregationGranularity, request.getAggregationGranularity());
+
+    Set<String> actualMetricName = request.getMetricNames();
+    if (metricNames == null) {
+      Assert.assertNull(actualMetricName);
+    } else {
+      Assert.assertEquals(metricNames.size(), actualMetricName.size());
+      for (String name : metricNames) {
+        Assert.assertTrue(actualMetricName.contains(name));
+      }
+    }
+
+    Map<String, Set<String>> actualComponentInstance = request.getComponentNameInstanceId();
+    if (componentNameInstanceId == null) {
+      Assert.assertNull(actualComponentInstance);
+    } else {
+      Assert.assertEquals(componentNameInstanceId.size(), actualComponentInstance.size());
```

Comment:
Just a nit, but doing a static import of `Assert.assertFoo` methods will help make the code less verbose.